### PR TITLE
@W-11443022: Changes to allow user to choose privileged access

### PR DESCRIPTION
**Pull request by npattanayak**

### CODE REVIEW

File: pkg/sloop/ingress/kubeclient.go [hunk 1]
```diff
@@ -12,25 +12,27 @@ import (
...
+	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/tools/clientcmd/api"
 )
 
 // GetKubernetesContext takes optional user preferences and returns the Kubernetes context in use
-func GetKubernetesContext(masterURL string, kubeContextPreference string) (string, error) {
+func GetKubernetesContext(masterURL string, kubeContextPreference string, privilegedAccess bool) (string, error) {
 	glog.Infof("Getting k8s context with user-defined config masterURL=%v, kubeContextPreference=%v.", masterURL, kubeContextPreference)
+	contextInUse := kubeContextPreference
```
Comment (sana-jawad) [inline pkg/sloop/ingress/kubeclient.go:23]: It seems unnecessary as the value is assigned anyways on line 32

Comment (npattanayak) [reply pkg/sloop/ingress/kubeclient.go:23]: Reason I have initialized that way is that : If privileged access not given then it directly assing kubecontextpreference

**Review by sana-jawad — COMMENTED** [review]
Few minor comments

File: pkg/sloop/ingress/kubeclient.go [hunk 2]
```diff
@@ -55,7 +65,7 @@ func MakeKubernetesClient(masterURL string, kubeContext string) (kubernetes.Inte
 		return nil, err
 	}
 
-	glog.Infof("Created k8sclient with context=%v, masterURL=%v, configFile=%v.", kubeContext, config.Host, clientConfig.ConfigAccess().GetLoadingPrecedence())
+	glog.Infof("Created k8sclient with above configurations")
```
[the remark this answers is not in the document]
Comment (npattanayak) [reply pkg/sloop/ingress/kubeclient.go:74]: In the earlier info,  the log statement has one parameter of client Config too, however now we have 2 loops one which creats client config and then takes clientconfig() function , other which directly uses buildfromflags. Thats why I have shifter their logs to their respective loops and added a common log to show the creation is complete. Another work around could be removing the `clientConfig.ConfigAccess().GetLoadingPrecedence()` from logging and mentioning just rest 2.

File: pkg/sloop/ingress/kubeclient_test.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	if !methodInvoked {
+		t.Errorf("RawConfig not invoked")
+	}
+}
+
+func TestGetKubernetesContextNoPrivilegedAccess(t *testing.T) {
+	var context string
+	context, _ = GetKubernetesContext("", "", false)
+	assert.Equal(t, context, "")
+
```
[the remark this answers is not in the document]
Comment (npattanayak) [reply pkg/sloop/ingress/kubeclient_test.go]: Noted

File: pkg/sloop/ingress/kubeclient_test.go [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+			t.Errorf("Got unexpected flag")
+		}
+		return &rest.Config{}, nil
+	}
+	MakeKubernetesClient("", "", privilegedAccess)
+	if !methodInvoked {
+		t.Errorf("ClientConfig not invoked")
+	}
+}
+func TestMakeKubernetesClientByBuildConfigFromFlag(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (npattanayak) [reply pkg/sloop/ingress/kubeclient_test.go:55]: Noted

File: pkg/sloop/ingress/kubeclient.go [hunk 5]
```diff
@@ -12,25 +12,27 @@ import (
...
 	_ "k8s.io/client-go/plugin/pkg/client/auth/azure"
 	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"k8s.io/client-go/rest"
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/tools/clientcmd/api"
 )
 
 // GetKubernetesContext takes optional user preferences and returns the Kubernetes context in use
-func GetKubernetesContext(masterURL string, kubeContextPreference string) (string, error) {
+func GetKubernetesContext(masterURL string, kubeContextPreference string, PrivilegedAccess bool) (string, error) {
```
Comment (sana-jawad) [inline pkg/sloop/ingress/kubeclient.go]: please do not capitalize it, it makes it global variable

Comment (npattanayak) [reply pkg/sloop/ingress/kubeclient.go]: Hi @sana-jawad The review is at 35 commits, could you please review changes from all commits. This was taken up earlier .

File: pkg/sloop/ingress/kubeclient.go (hunk 1 above)
Comment (npattanayak) [reply pkg/sloop/ingress/kubeclient.go:29]: @sana-jawad  This is the latest file

**Review by sana-jawad — APPROVED** [review]

**Pull request merged**